sql/src/main/java/io/druid/sql/calcite/rule/SelectRules.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.sql.calcite.rule;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+import io.druid.query.dimension.DefaultDimensionSpec;
+import io.druid.query.dimension.DimensionSpec;
+import io.druid.query.dimension.ExtractionDimensionSpec;
+import io.druid.query.extraction.ExtractionFn;
+import io.druid.query.groupby.orderby.DefaultLimitSpec;
+import io.druid.query.groupby.orderby.OrderByColumnSpec;
+import io.druid.segment.column.Column;
+import io.druid.segment.column.ValueType;
+import io.druid.sql.calcite.expression.Expressions;
+import io.druid.sql.calcite.expression.RowExtraction;
+import io.druid.sql.calcite.rel.DruidRel;
+import io.druid.sql.calcite.rel.SelectProjection;
+import io.druid.sql.calcite.table.RowSignature;
+import org.apache.calcite.plan.RelOptRule;
+import org.apache.calcite.plan.RelOptRuleCall;
+import org.apache.calcite.rel.core.Project;
+import org.apache.calcite.rel.core.Sort;
+
+import java.util.List;
+
+public class SelectRules
+{
+  private SelectRules()
+  {
+    // No instantiation.
+  }
+
+  public static List<RelOptRule> rules()
+  {
+    return ImmutableList.of(
+        new DruidSelectProjectionRule(),
+        new DruidSelectSortRule()
+    );
+  }
+
+  static class DruidSelectProjectionRule extends RelOptRule
+  {
+    private DruidSelectProjectionRule()
+    {
+      super(operand(Project.class, operand(DruidRel.class, none())));
+    }
+
+    @Override
+    public void onMatch(RelOptRuleCall call)
+    {
+      final Project project = call.rel(0);
+      final DruidRel druidRel = call.rel(1);
+
+      if (druidRel.getQueryBuilder().getSelectProjection() != null
+          || druidRel.getQueryBuilder().getGrouping() != null
+          || druidRel.getQueryBuilder().getLimitSpec() != null) {
+        return;
```

Comment:
How about implementing ```public boolean matches(RelOptRuleCall call)```? I think this will be a better approach for avoiding not-matched rules early. 
I know these rule classes are just moved to here, but it will be good if we can improve them.